proposals/pattern-const-expressions.md

```diff
@@ -0,0 +1,97 @@
+# Constant `is` pattern expressions
+
+* [x] Proposed
+* [ ] Prototype: None
+* [ ] Implementation: None
+* [ ] Specification: Started, below
+
+## Summary
+[summary]: #summary
+
+Consider `is` pattern expressions as constant if the LHS is constant.
+
+## Motivation
+[motivation]: #motivation
+
+When the pattern expressions are converted into simpler boolean expressions using equality and comparison operators, they are considered constant expressions if all parts of the expression are considered constant and can be evaluated during compilation. Also, the expressions returned from the ternary operator can also be considered constant if all sides are constant (condition, consequence and alternative).
```

Comment:
Not sure what the ternary-operator clause means here.

Reply:
The `?:` conditional operator, I'm used to calling it the ternary operator and forget its actual name sometimes. Will adjust the doc accordingly.

Reply:
that's not what i mean.  What i'm saying is: i don't see how that sentence helps here.  WHy are we bringing up conditional expressions?  Are you intending to change how they work?

Reply:
I see now, the only reason why this is being mentioned is that now certain `is` expressions will be able to be used in the condition of a conditional operator, and the result would then be able to be considered constant. In other words, `a is 3 ? 3 : 4` will now be considered constant, since the underlying `a is 3` expression is also constant. More of a side effect without explicitly touching that area.

Reply:
i would leave it out from here.  You can show it as an example later on :)